blockchain/block.go

```diff
@@ -48,10 +50,18 @@ func (bh *BlockHeader) Marshal() []byte {
 	buf = append(buf, bh.Target.Marshal()...)
 	buf = append(buf, tempBufTime...)
 	buf = append(buf, tempBufNonce...)
+	buf = append(buf, bh.ExtraData...)
 
 	return buf
 }
 
+// Len returns the length in bytes of the BlockHeader.
+func (bh *BlockHeader) Len() int {
+	l := 2*(32/8) + 64/8 + 2*HashLen
```

Comment:
`unsafe` doesn't seem ideal, but isn't there some notion of `sizeof` in golang?https://stackoverflow.com/questions/26975738/how-to-get-memory-size-of-variable